interfaces/acados_template/acados_template/mpc_utils.py

```diff
@@ -144,13 +144,8 @@ def detect_constraint_structure(model: AcadosModel, constraints: AcadosOcpConstr
     if model.p_global is None:
         p_global = []  # to have same structure of model.p
 
-    if isinstance(x, ca.SX):
-        casadi_var = ca.SX
-    elif isinstance(x, ca.MX):
-        casadi_var = ca.MX
-    else:
-        raise TypeError(
-            "Optimization variables 'x' must be of type 'casadi SX' or 'casadi MX'")
+    casadi_var = model.get_casadi_symbol()
+    casadi_dm_zeros = ca.DM.zeros
```

Comment:
I used `get_casadi_symbol()` for creating symbols but for the zeros I needed casadi DM so I did it as above